libbeat/template/template.go

```diff
@@ -106,7 +106,13 @@ func (t *Template) generate(properties common.MapStr, dynamicTemplates []common.
 		"settings": common.MapStr{
 			"index.refresh_interval": "5s",
 		},
-		"template": t.GetName() + "-*",
+	}
+
+	// ES 6 moved from template to index_patterns: https://github.com/elastic/elasticsearch/pull/21009
+	if t.esVersion.major >= 6 {
```

Comment:
We should probably enhance `version` in the future to also support `GreaterThen`functions or similar. This will do for the moment.